src/app/domain/geolocation/helpers/quota.ts

```diff
@@ -0,0 +1,118 @@
+import { differenceInDays } from 'date-fns'
+
+import { Q } from 'cozy-client'
+import CozyClient from 'cozy-client'
+import flag from 'cozy-flags'
+import Minilog from 'cozy-minilog'
+
+import { showLocalNotification } from '/libs/notifications/notifications'
+import { t } from '/locales/i18n'
+import { getData, storeData, StorageKeys } from '/libs/localStore/storage'
+
+const MAX_DAYS_TO_CAPTURE_UNLIMITED = -1
+const ONE_DAY = 24 * 60 * 60 * 1000
+
+const log = Minilog('📍 Geolocation')
+
+export interface FirstTimeserie {
+  id: string
+  _id: string
+  type: string
+  startDate: string
+}
+
+const isMaxDaysToCaptureInvalid = (
+  maxDaysToCapture: number | null
+): maxDaysToCapture is null => {
+  return typeof maxDaysToCapture !== 'number'
+}
+
+const isMaxDaysToCaptureUnlimited = (maxDaysToCapture: number): boolean => {
+  return maxDaysToCapture === MAX_DAYS_TO_CAPTURE_UNLIMITED
+}
+
+const getFirstTimeserie = async (
+  client: CozyClient
+): Promise<FirstTimeserie | undefined> => {
+  try {
+    const firstTimeserieCachedLocally = await getData<FirstTimeserie>(
+      StorageKeys.FirstTimeserie
+    )
+
+    if (firstTimeserieCachedLocally) {
+      return firstTimeserieCachedLocally
+    } else {
+      const { data } = (await client.fetchQueryAndGetFromState({
+        definition: Q('io.cozy.timeseries.geojson')
+          .where({ _id: { $gt: null } })
+          .select(['_id', 'startDate'])
```

Comment:
You don't need the `_id`, it will be returned anyway. Then, you can remove it from the `.select` and adapt the selector: ` .where({ startDate: { $gt: null } })`

Reply:
Can't we keep the `_id` in the `select`? 

1/ It's more explicit 
2/ Will be easier to have type checking later

Reply:
Actually I might confuse the files behaviour, where the stack force some fields: https://github.com/cozy/cozy-stack/blob/master/web/files/files.go#L1488
I see nothing like this for other doctypes: https://github.com/cozy/cozy-stack/blob/master/web/data/data.go#L347
So if you need the `_id`, you actually need to add it in the select

Reply:
I just updated `.where({ startDate: { $gt: null } })` . Thanks!